### ISSUE TITLE
Invalidating file with syntax error breaks further invalidations for that file

### ISSUE THREAD

**Issue by Bnaya** [jackfranklin/jspm-dev-builder#15]
You use incremental build
The initial build succeed and then change file to have a syntax error,
the file is sent to builder.build(filepath) 
the build fails with error like:

```
jspm build jspm build error file:///Users/bnaya/dev/***/mobile2/src/js/***/Mobile/Components/Root.jsx: Unterminated string constant (55:0)
  53 |     }
  54 | }
> 55 | '
     | ^
  56 | export default Root;
  57 |  Error on translate for js/***/Mobile/Components/Root.jsx at file:///Users/bnaya/dev/***/mobile2/src/js/***/Mobile/Components/Root.jsx
    Loading js/***/Mobile/main.js
    SyntaxError: file:///Users/bnaya/dev/***/mobile2/src/js/***/Mobile/Components/Root.jsx: Unterminated string constant (55:0)
  53 |     }
  54 | }
> 55 | '
     | ^
  56 | export default Root;
  57 | 
    at Parser.pp.raise (evalmachine.<anonymous>:63062:13)
```

You fix the syntax error, and the watcher send the file to builder.build(filepath) again
But then you get that error:

```
jspm build jspm invalidation error Nothing has been invalidated for js/***/Mobile/Components/Root.jsx. It's likely that the module does not exist. If the module is loaded using a plugin, you need to append ! to its name in order to invalidate it properly
jspm build jspm build starting js/***/Mobile/main.js
jspm build Building non-SFX bundle
jspm build jspm build error file:///Users/bnaya/dev/***/mobile2/src/js/***/Mobile/Components/Root.jsx: Unterminated string constant (55:0)
  53 |     }
  54 | }
> 55 | '
     | ^
  56 | export default Root;
  57 |  Error on translate for js/***/Mobile/Components/Root.jsx at file:///Users/bnaya/dev/***/mobile2/src/js/***/Mobile/Components/Root.jsx
  Loading js/***/Mobile/main.js
  SyntaxError: file:///Users/bnaya/dev/***/mobile2/src/js/***/Mobile/Components/Root.jsx: Unterminated string constant (55:0)
  53 |     }
  54 | }
> 55 | '
     | ^
  56 | export default Root;
  57 | 
    at Parser.pp.raise (evalmachine.<anonymous>:63062:13)
...
```

It's like systemjs/builder caches the broken content of the file, and the builder.invalidate fails like its not in the cache.

And need to restart you watch task to make it work again.

**Comment by jackfranklin** [comment]
Yup, I believe this is the same as https://github.com/jackfranklin/jspm-dev-builder/issues/11. I've got a plan but haven't been able to find the time so far.

**Comment by Bnaya** [comment]
I think its a bug inside systemjs builder, probably something with caching.
I've started to debug systemjs builder i hope i will find something

**Comment by Bnaya** [comment]
I've fixed that issue with systemjs/builder#439

**Comment by Bnaya** [comment]
Should be resolved with the next release of jspm!

**Comment by jackfranklin** [comment]
@Bnaya awesome!

**Comment by Bnaya** [comment]
Fixed in jspm 0.16.20
